Soleras3/react-challenge-project-jan-2023
# Fix Order Time formatting

## Changes
- Fix the issue by refactoring date string formatting to `.toLocaleTimeString`. 

## Purpose
Order times could display minutes or second as single digit (ex: `18:9:2`). See: https://github.com/Shift3/react-challenge-project-jan-2023/issues/2

## Approach
Dates and times are challenging. Always prefer a native system implementation, or a library, when handling them-- the bugs are already fixed, and they provide often necessary locale support.

Fix's format is in 24h style but the user requirement is unclear; the issue only specifies `hh:mm:dd` format. If the default is preferred (12h in the US, 24h many other locales) it's easy to change.

## Pre-Testing TODOs
Requires: https://github.com/Soleras3/react-challenge-project-jan-2023/pull/2

- Create an order where the minute or second is less than 10. Ex: 12:09:05.
- Or: create an order then modify its `createdAt` in the database with a test-valid minute or second.

## Testing Steps
1. Login and navigate to View Orders.
2. Observe that order time per order properly displays `hh:mm:ss`.
![Screen Shot 2023-03-16 at 9 18 12 PM](https://user-images.githubusercontent.com/76075167/225795546-e6c200da-f76f-4a9d-bf44-9cd2596112cf.png)


## Learning
Javascript's native solution was researched through mozilla's [toLocaleTimeString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleTimeString) doc page.
Closes https://github.com/Shift3/react-challenge-project-jan-2023/issues/2
